private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/gradle/GradleResolver.java

```diff
@@ -427,6 +430,25 @@ private Path getGradleInstallationPath() {
     }
   }
 
+  private Path getPersistentGradleHomeForRepo() {
+    // We check for BUILD_WORKSPACE_DIRECTORY which will be set for most usages
+    // with bazel run. It won't be available with tests, so we fall back to TEST_SRCDIR
+    // which will map to the root of the runfiles tree
+    String workspaceRoot =
+        Optional.ofNullable(System.getenv("BUILD_WORKSPACE_DIRECTORY"))
+            .orElse(System.getenv("TEST_SRCDIR"));
+
+    // If none are set, just return null so we fall back to the isolated cache
+    if (workspaceRoot == null) return null;
```

Comment:
Please always use braces, even for one line blocks. I've seen too many bugs caused by people not realising that a block had already ended.